src/ts/components/button.ts

```diff
@@ -27,16 +32,29 @@ export class Button<Config extends ButtonConfig> extends Component<Config> {
 
     this.config = this.mergeConfig(config, {
       cssClass: 'ui-button',
+      role: 'button',
+      tabindex: 0,
     } as Config, this.config);
   }
 
   protected toDomElement(): DOM {
-    // Create the button element with the text label
-    let buttonElement = new DOM('button', {
-      'type': 'button',
+    const buttonElementAttributes: { [name: string]: string } = {
       'id': this.config.id,
+      'aria-label': i18n.performLocalization(this.config.ariaLabel),
       'class': this.getCssClasses(),
-    }).append(new DOM('span', {
+      /**
+      * WCAG20 standard to display if a button is pressed or not
+      */
+      'aria-pressed': 'false',
+      'tabindex': this.config.tabindex.toString(),
+    };
+
+    if (this.config.role != null) {
+      buttonElementAttributes['role'] = this.config.role;
+    }
```

Comment:
Is the `if` necessary? Can't you simply assign the `role` whether it is set or not?

Reply:
If the role is not defined (not needed in a specific case) the attribute will be still added to the element with empty value